tests/units/DB.php

```diff
@@ -497,4 +497,38 @@ function () use ($db, $create_query_template, $drop_query_template) {
          }
       )->error()->notExists();
    }
+
+   public function testSavepoints() {
+      global $DB;
+
+      $DB->beginTransaction();
+      try {
```

Comment:
CI is not happy with the empty catch; and I guess we should not try/catch here; any expected exception can be tested with atoum

Reply:
The problem is unexpected exceptions. As far as I could tell, there is no automatic rollback in this test suite so if something fails, it could be auto-committed to the test DB.

Reply:
I do not get the point. You first start a new transaction, so unless you explicitly call a commit, all will be rolled back per default.

Reply:
Nevermind. I misunderstood the MySQL documentation.